azure-quantum/azure/quantum/qiskit/backends/microsoft.py

```diff
@@ -0,0 +1,75 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+
+from typing import TYPE_CHECKING
+from azure.quantum.version import __version__
+
+from .backend import AzureBackend
+
+from qiskit.providers.models import BackendConfiguration
+from qiskit.providers import Options
+from qiskit_qir import SUPPORTED_INSTRUCTIONS as QIR_BASIS_GATES
+
+if TYPE_CHECKING:
+    from azure.quantum.qiskit import AzureQuantumProvider
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "MicrosoftBackend", "MicrosoftResourcesEstimatorBackend"
+]
+
+class MicrosoftBackend(AzureBackend):
+    """Base class for interfacing with a Microsoft backend in Azure Quantum"""
+
+    @classmethod
+    def _default_options(cls):
+        return Options(entryPoint="main", arguments=[], targetCapibility="BasicExecution")
+
+    @classmethod
+    def _azure_config(cls):
+        return {
+            "blob_name": "inputData",
+            "content_type": "qir.v1",
+            "provider_id": "microsoft.simulator",
+            "input_data_format": "qir.v1",
+            "output_data_format": "microsoft.resource-estimates.v1",
+            "to_qir_kwargs": {"record_output": False, "use_static_qubit_alloc": False, "use_static_result_alloc": False}
+        }
+
+class MicrosoftResourcesEstimatorBackend(MicrosoftBackend):
```

Comment:
We may want to rename this to MicrosoftResourceEstimationBackend singular (instead of plural Resources)